Add composer installation to docs/welcome

racmanuel opened the pull request:
Hello, I have added how to install Secure Custom Fields in a plugin or theme using composer.

racmanuel committed:
Add composer installation to docs/welcome

racmanuel committed:
Merge branch 'WordPress:trunk' into trunk

racmanuel committed:
Merge branch 'trunk' into trunk

priethor commented on docs/welcome/installation.md:
````diff
@@ -27,6 +27,76 @@ Before installing, ensure your site meets these requirements:
+
+```php
+// Hide the SCF admin menu item.
+add_filter( 'acf/settings/show_admin', '__return_false' );
````
We should be careful not to suggest always disabling the admin menu: what if the site using the plugin/theme uses SCF anyway?

racmanuel replied:
@priethor You're right, I think this should be an optional step, I'll modify it in a new commit.

priethor replied:
The same goes for plugin updates. I think it should be checked whether SCF is already installed with `get_plugins()`  before trying to hide either.

priethor commented on docs/welcome/installation.md:
```diff
@@ -27,6 +27,76 @@ Before installing, ensure your site meets these requirements:
 
+### Composer Installation
+
+This guide explains how to install and integrate the **Secure Custom Fields** plugin in your WordPress theme or plugin using Composer.
```
Could you add a bit more about the `why`? A one liner with an example should be enough.

racmanuel replied:
@priethor I made in the new commit and pr para añadir una section about : Why integrate Secure Custom Fields (SCF) with Composer.

priethor reviewed (changes requested):
Thanks for submitting this! I've left a couple of notes.

racmanuel committed:
docs: update SCF installation guide with Composer integration, benefits, and optional admin menu and update notification hiding

docs: update SCF installation guide with Composer integration, benefits, and optional admin menu and update notification hiding

racmanuel committed:
Merge branch 'trunk' into trunk

racmanuel committed:
Merge branch 'trunk' into trunk

racmanuel committed:
Merge branch 'trunk' into trunk

racmanuel committed:
Merge branch 'trunk' into trunk

priethor commented on docs/welcome/installation.md:
```diff
@@ -26,11 +26,132 @@ Before installing, ensure your site meets these requirements:
+    }
+  },
+  "require": {
+    "wpackagist-plugin/secure-custom-fields": "6.4.1"
```
```suggestion
    "wpackagist-plugin/secure-custom-fields": "^6.4"
```

Otherwise, it will always install that specific version; let's not take for granted that readers are familiar with Composer 🙂

priethor commented on docs/welcome/installation.md:
```diff
@@ -26,11 +26,132 @@ Before installing, ensure your site meets these requirements:
+- If you are bundling SCF inside your plugin or theme and want to **control all the custom fields yourself** without allowing clients or users to modify them.
+- If you want to **maintain full control** over SCF versions and updates to avoid compatibility issues caused by manual updates.
+
+> **Note:** Hiding updates means you are responsible for manually updating SCF when necessary to keep your project secure and compatible.
```
```suggestion
> **Note:** Hiding updates means you are responsible for manually updating SCF when necessary to keep your project secure and compatible, but it also helps avoid potential conflicts between Composer and the built-in updater.
```

racmanuel committed:
Update docs/welcome/installation.md

let's not take for granted that readers are familiar with Composer

Co-authored-by: Héctor <27339341+priethor@users.noreply.github.com>

racmanuel committed:
Update docs/welcome/installation.md

avoid potential conflicts between Composer and the built-in updater

Co-authored-by: Héctor <27339341+priethor@users.noreply.github.com>

racmanuel committed:
Merge branch 'trunk' into trunk

racmanuel committed:
Merge branch 'trunk' into trunk
